### PR TITLE
Added `AvgPool1D` + stateful test

### PR DESCRIPTION
**Pull request by mobley-trent**

### CODE REVIEW

**Comment by mobley-trent** [comment]
Closes #15348

**Comment by mobley-trent** [comment]
@zaeemansari70 do you mind reviewing this PR ? The tests are passing on my end

**Review by MahmoudAshraf97 — CHANGES_REQUESTED** [review]

File: ivy/stateful/layers.py [hunk 1]
```diff
@@ -1543,3 +1543,58 @@ def _forward(self, inputs):
...
+        self,
+        kernel_size,
+        stride,
+        padding,
+        /,
+        *,
+        data_format="NWC",
+        device=None,
+        v=None,
+        dtype=None,
```
[the remark this answers is not in the document]
Comment (mobley-trent) [reply ivy/stateful/layers.py:1559]: They are needed in the `__init__` method of the `Module` class. I followed the convention used in the rest of the layers

**Comment by mobley-trent** [comment]
> please review your implementation and remove unnecessary arguments, you can check `ivy.ReLU` class for example

The implementation is consistent with the rest of the layers in the module, even `AvgPool2D` which is right next to it

**Pull request closed without merging**